zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -106,7 +102,12 @@ boolean setDomainAuthUpdater(ConnectionAuthInfoUpdater updater) {
    * Install a persistent recursive watch on the root path.
    */
   private void addWatches() {
-    zks.getZKDatabase().addWatch(rootPath, this, ZooDefs.AddWatchModes.persistentRecursive);
+    try {
+      zks.getZKDatabase().getDataTree().getChildren(rootPath, new Stat(), this::processEvent);
+    } catch (KeeperException.NoNodeException e) {
+      throw new IllegalStateException(
+          "Client URI domain mapping root path config is not set!");
```

Comment:
It could also mean the root path config is set, but the znode is not yet created right?